gtda/externals/python/ripser_interface.py

```diff
@@ -175,7 +176,10 @@ def ripser(X, maxdim=1, thresh=np.inf, coeff=2, metric="euclidean",
         r_cover = lambdas[-1]
         dm = dperm2all[:, idx_perm]
     else:
-        dm = pairwise_distances(X, metric=metric)
+        if metric == 'precomputed':
+            dm = X
+        else:
+            dm = pairwise_distances(X, metric=metric)
```

Comment:
In this way, we would still error on input with negative entries, due to https://github.com/scikit-learn/scikit-learn/blob/95d4f0841d57e8b5f6b2a570312e9d832e69debc/sklearn/metrics/pairwise.py#L1721. One of the main points of this PR is to avoid this.